src/wallet/rpcwallet.cpp

```diff
@@ -1500,6 +1500,267 @@ UniValue listtransactions(const JSONRPCRequest& request)
+                }
+                continue;
+            }
+            // check for remaining when new entry exists
+            if ((*pRemaining) == 0) {
+                return false;
+            }
+            (*pRemaining)--;
+            UniValue entry(UniValue::VOBJ);
+            if (involvesWatchonly || (::IsMine(*pwallet, r.destination) & ISMINE_WATCH_ONLY)) {
+                entry.pushKV("involvesWatchonly", true);
+            }
+            MaybePushAddress(entry, r.destination);
+            if (wtx.IsCoinBase())
+            {
+                if (wtx.GetDepthInMainChain(locked_chain) < 1)
+                    entry.pushKV("category", "orphan");
+                else if (wtx.IsImmatureCoinBase(locked_chain))
+                    entry.pushKV("category", "immature");
+                else
+                    entry.pushKV("category", "generate");
+            }
+            else
+            {
+                entry.pushKV("category", "receive");
+            }
+            entry.pushKV("amount", ValueFromAmount(r.amount));
+            if (pwallet->mapAddressBook.count(r.destination)) {
+                entry.pushKV("label", label);
+            }
+            entry.pushKV("vout", r.vout);
+            if (fLong)
+                WalletTxToJSON(pwallet->chain(), locked_chain, wtx, entry);
+            ret.push_back(entry);
+        }
+    }
+
+    return true;
+}
+
+UniValue listtransactions2(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    std::string help_text =
+      "listtransactions2 (label count nextpagepointer include_watchonly)\n"
+            "\nIf a label name is provided, this will return only incoming transactions paying to addresses with the specified label.\n"
+            "\nReturns up to 'count' most recent transactions from tip of entries or nextpagepointer (exclusive).\n"
+            "\nArguments:\n"
+            "1. \"label\"    (string, optional) If set, should be a valid label name to return only incoming transactions\n"
+            "              with the specified label, or \"*\" to disable filtering and return all transactions.\n"
+            "2. count             (numeric, optional, default=10) The number of transactions to return\n"
+            "3. nextpagepointer         (string, optional, default=null) Continue the existing listing by passing last known txid:vout.\n"
+            "5. include_watchonly (bool, optional, default=false) Include transactions to watch-only addresses (see 'importaddress')\n"
+            "\nResult:\n"
+            "[\n"
+            "  {\n"
+            "    \"address\":\"address\",    (string) The bitcoin address of the transaction.\n"
+            "    \"category\":\"send|receive\", (string) The transaction category.\n"
```

Comment:
There are more categories than just `send` and `receive`. See https://github.com/bitcoin/bitcoin/pull/14653

Reply:
I was using branch 0.17 , and when i upgraded to master. help file was from the old one. will fix

Reply:
I see original `lsittransactions` only contains `send|receive`. @andrewtoth Please inform me. What are the other values. I guess there's `internal`.

Reply:
The other 3 are generate, immature and orphan.

Reply:
okay. I have made the change. i will add it after review.